engine/apps/alerts/tasks/unsilence.py

```diff
@@ -30,8 +30,14 @@ def unsilence_task(alert_group_pk):
             )
             return
         if alert_group.status == AlertGroup.SILENCED and alert_group.is_root_alert_group:
+            initial_state = alert_group.state
             task_logger.info(f"unsilence alert_group {alert_group_pk} and start escalation if needed")
+
             alert_group.un_silence()
+            # update metrics
```

Comment:
Maybe it's better to update metrics in `AlertGroup.un_silence`?

Reply:
Updating metrics inside the state transition methods is tricky because you may lose track of the original state in some cases (e.g. we call un_silence in multiple places, for example we may call this during acknowledge), and also because bulk updates don't go through these methods.